File: src/main/java/org/springframework/samples/ntfh/enemy/HordeEnemyType.java
package org.springframework.samples.ntfh.enemy;

import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.Table;

import lombok.Getter;

@Getter
@Table(name = "horde_enemy_types")
public class HordeEnemyType {
    @Enumerated(EnumType.STRING)
    private HordeEnemyTypeEnum hordeEnemyTypeEnum;

    @Enumerated(EnumType.STRING)
    private HordeModifierTypeEnum hordeEnemyModifierTypeEnum;

    @Enumerated(EnumType.STRING)
    private Integer endurance;

}

File: react/src/interfaces/HordeEnemyType.ts
import { HordeEnemyTypeEnum } from "../types/HordeEnemyTypeEnum";
import { HordeModifierTypeEnum } from "../types/HordeModifierTypeEnum";

export interface HordeEnemyType {
  hordeEnemyTypeEnum: HordeEnemyTypeEnum;
  hordeModifierTypeEnum: HordeModifierTypeEnum;
  endurance: number;
}
